src/core/meshcop/commissioner.cpp

```diff
@@ -287,25 +287,25 @@ otError Commissioner::AddJoiner(const Mac::ExtAddress *aEui64, const char *aPskd
     return error;
 }
 
-otError Commissioner::GetNextJoinerInfo(uint16_t &aIterator, otJoinerInfo &aJoiner) const
+otError Commissioner::GetNextJoinerInfo(uint16_t *aIterator, otJoinerInfo *aJoiner) const
```

Comment:
Is there a reason for this change?
The public OT APIs should be C but inside core, we use C++ and should prefer reference over pointers (from style-guide).